pkg/testcase/upgradenodereplacement.go

```diff
@@ -0,0 +1,570 @@
+package testcase
+
+import (
+	"fmt"
+	"os"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/rancher/distros-test-framework/config"
+	"github.com/rancher/distros-test-framework/factory"
+	"github.com/rancher/distros-test-framework/pkg/aws"
+	"github.com/rancher/distros-test-framework/shared"
+
+	. "github.com/onsi/ginkgo/v2"
+)
+
+const (
+	agent  = "agent"
+	master = "master"
+)
+
+func TestUpgradeReplaceNode(version string) error {
```

Comment:
humm sorry, i have too much places that needs error returned to a better visibilty and maintainence, so i would need to flood the code with 
Expect(error)nothave(occured). stuff 

whats the problem of having this expect on the entrypoint ? 
its not a conditional flow, is just an error check 

```
	It("Upgrade by Node replacement", func() {
		err := testcase.TestUpgradeReplaceNode(customflag.ServiceFlag.InstallMode.String())
		Expect(err).NotTo(HaveOccurred())
	})
```

Reply:
There's no problem from a usability perspective necessarily, but we've done all other test functions with Expect statements instead of returns. If each return has a very clear message of where and why it's failing, it would be fine but we would want to redesign the other functions for consistency in that way. In my opinion, it's best the way we currently have the other ones, where we say what we expect. 

It's also a little easier to read (from a "test as documentation" perspective) so instead of this:
```
	awsDependencies, err := aws.AddNode()
	if err != nil {
		return shared.ReturnLogError("error adding aws nodes: %w\n", err)
	}
```
It'd just be:
```
	awsDependencies, err := aws.AddNode()
        Expect(err).To(BeNil(), "error adding aws nodes")
```

Reply:
i really dont like ( never did ) this design dependent on an external lib and its terrible dealing errors with what 
but i must say that i totally agree with the design that is current in place, that we should follow it.

Done. test case func updated